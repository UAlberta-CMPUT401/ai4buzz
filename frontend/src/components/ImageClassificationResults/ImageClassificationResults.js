import React from 'react';
import styles from './ImageClassificationResults.module.css';

const ImageClassificationResults = ({ results }) => {
  return (
    <div>
      {Object.keys(results).map((object, idx) => {
        return (
          <div key={idx} className={styles.imageClassificationResultsData}>
            <div>{object}</div>
            <div className="value">{(results[object] * 100).toFixed(3)}%</div>
          </div>
        );
      })}
    </div>
  );
};

export default ImageClassificationResults;
