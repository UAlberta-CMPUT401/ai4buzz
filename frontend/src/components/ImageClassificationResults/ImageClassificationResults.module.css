.imageClassificationResultsData {
  display: flex;
  justify-content: space-between;
}
