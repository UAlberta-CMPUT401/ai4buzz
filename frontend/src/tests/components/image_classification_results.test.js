import React from 'react';
import ImageClassificationResults from '../../components/ImageClassificationResults/ImageClassificationResults';
import renderer from 'react-test-renderer';

it('Renders ImageClassification component correctly', () => {
  const imageClassificationResults = {
    'web site': 0.9249883890151978,
    'book jacket': 0.01968519203364849,
    'comic book': 0.012182716280221939,
    envelope: 0.0011725202202796936,
    'analog clock': 0.0009474915568716824,
  };
  const tree = renderer
    .create(<ImageClassificationResults results={imageClassificationResults} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
