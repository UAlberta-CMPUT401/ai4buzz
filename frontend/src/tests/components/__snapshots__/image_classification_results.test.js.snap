// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ImageClassification component correctly 1`] = `
<div>
  <div
    className="imageClassificationResultsData"
  >
    <div>
      web site
    </div>
    <div
      className="value"
    >
      92.499
      %
    </div>
  </div>
  <div
    className="imageClassificationResultsData"
  >
    <div>
      book jacket
    </div>
    <div
      className="value"
    >
      1.969
      %
    </div>
  </div>
  <div
    className="imageClassificationResultsData"
  >
    <div>
      comic book
    </div>
    <div
      className="value"
    >
      1.218
      %
    </div>
  </div>
  <div
    className="imageClassificationResultsData"
  >
    <div>
      envelope
    </div>
    <div
      className="value"
    >
      0.117
      %
    </div>
  </div>
  <div
    className="imageClassificationResultsData"
  >
    <div>
      analog clock
    </div>
    <div
      className="value"
    >
      0.095
      %
    </div>
  </div>
</div>
`;
